ingestion/src/metadata/ingestion/source/messaging/kafka.py

```diff
@@ -34,57 +29,41 @@
 from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
     OpenMetadataConnection,
 )
-from metadata.generated.schema.entity.services.messagingService import (
-    MessagingService,
-    MessagingServiceType,
-)
 from metadata.generated.schema.metadataIngestion.workflow import (
     Source as WorkflowSource,
 )
 from metadata.generated.schema.type.entityReference import EntityReference
 from metadata.ingestion.api.common import logger
 from metadata.ingestion.api.source import InvalidSourceException, Source, SourceStatus
 from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.source.messaging.messaging_service import MessagingServiceSource
 from metadata.utils.connection_clients import KafkaClient
 from metadata.utils.connections import get_connection, test_connection
 from metadata.utils.filters import filter_by_topic
 
 
 @dataclass
-class KafkaSourceStatus(SourceStatus):
-    topics_scanned: List[str] = field(default_factory=list)
-    filtered: List[str] = field(default_factory=list)
+class KafkaTopicDetails:
+    """
+    Wrapper Class to combine the topic_name with topic_metadata
+    """
 
-    def topic_scanned(self, topic: str) -> None:
-        self.topics_scanned.append(topic)
+    topic_name: str
+    topic_metadata: Any
 
-    def dropped(self, topic: str) -> None:
-        self.filtered.append(topic)
+    def __init__(self, topic_name: str, topic_metadata: Any) -> None:
+        self.topic_name = topic_name
+        self.topic_metadata = topic_metadata
```

Comment:
Should that be defined as a jsonschema or at least using pydantic for consistency?

Reply:
@TeddyCr Using pydantic..will update in the next commit.